ComponentKit/Components/Mac/CKMTableComponent.mm

```diff
@@ -0,0 +1,382 @@
+{
+  return (CKMTableComponent *)self.component;
+}
+
+static bool enableReloadSelection = true;
+
+static CKTransactionalComponentDataSourceChangeset *replaceItems(NSArray *oldModels, NSArray *models, NSIndexSet *oldSelection, NSIndexSet *newSelection) {
+  if (oldModels == models || [oldModels isEqualToArray:models]) {
+
+    // Ok, models are the same, but selection could have changed
+    if (enableReloadSelection && !(oldSelection == newSelection || [newSelection isEqualToIndexSet:oldSelection])) {
+      NSIndexSet *needsReload = symmetricDifference(oldSelection ?: [NSIndexSet indexSet], newSelection ?: [NSIndexSet indexSet]);
+
+      NSMutableDictionary *reload = [NSMutableDictionary dictionary];
+      [models enumerateObjectsAtIndexes:needsReload options:0 usingBlock:^(id  _Nonnull model, NSUInteger idx, BOOL * _Nonnull stop) {
+        reload[[NSIndexPath indexPathForItem:idx inSection:0]] = model;
+      }];
+
+      return [[[CKTransactionalComponentDataSourceChangesetBuilder transactionalComponentDataSourceChangeset] withUpdatedItems:reload] build];
+    } else if (!enableReloadSelection) {
+      return [[CKTransactionalComponentDataSourceChangesetBuilder transactionalComponentDataSourceChangeset] build];
+    }
+
+    return nil;
+  }
+
+  // Kinda gross to do this (should diff them and only emit a subset instead)
+  NSMutableSet *removedSet = [NSMutableSet set];
+  for (NSUInteger i=0; i < oldModels.count; i++) {
+    [removedSet addObject:[NSIndexPath indexPathForItem:i inSection:0]];
+  }
+  NSMutableDictionary *insert = [NSMutableDictionary dictionary];
+  NSInteger idx = 0;
+  for (id model : models) {
+    insert[[NSIndexPath indexPathForItem:idx inSection:0]] = model;
+    idx++;
+  }
+
+  return [[[[CKTransactionalComponentDataSourceChangesetBuilder transactionalComponentDataSourceChangeset] withRemovedItems:removedSet] withInsertedItems:insert] build];
+}
+
+static CKTransactionalComponentDataSourceChangeset *insertItems(NSArray *models, NSUInteger startIndex) {
+  NSMutableDictionary *insert = [NSMutableDictionary dictionary];
+  NSInteger idx = startIndex;
+  for (id model : models) {
+    insert[[NSIndexPath indexPathForItem:idx inSection:0]] = model;
+    idx++;
+  }
+  return [[[CKTransactionalComponentDataSourceChangesetBuilder transactionalComponentDataSourceChangeset] withInsertedItems:insert] build];
+};
+
+// Use our NSTableView to tell the dataSource how to make layout the content in our cells
+- (CKSizeRange)sizeRangeForDataSource
+{
+  NSScrollView *sv = (NSScrollView *)self.view;
+  CGFloat rightInset = sv.verticalScroller.bounds.size.width;
+  CGFloat contentWidth = _tableView.bounds.size.width - rightInset;
+
+  return {
+    {contentWidth, 0.0},
+    {contentWidth, HUGE_VALF},
+  };
+}
+
+CKComponentDelegateForwarder *appendSelectorsToForwarder(CKComponentDelegateForwarder *existing, CKComponentForwardedSelectors selectors)
+{
+  // Union our selectors with those of the existing delegate
```

Comment:
It could be wise to not add selectors that are already present, but we should watch out for O(N^2)